src/lib.rs

```diff
@@ -30,1209 +106,244 @@ extern crate core;
 
 #[cfg(all(feature = "alloc", not(feature = "std"), not(test)))]
 use alloc::{string::String, vec::Vec};
-use core::convert::{Infallible, TryFrom};
-use core::{fmt, mem};
+use core::fmt;
 
+#[doc(inline)]
 pub use crate::primitives::checksum::Checksum;
-use crate::primitives::checksum::{self, PackedFe32};
+#[cfg(feature = "alloc")]
+use crate::primitives::decode::{
+    CheckedHrpstring, CheckedHrpstringError, ChecksumError, UncheckedHrpstring,
+};
+#[doc(inline)]
 pub use crate::primitives::gf32::Fe32;
-use crate::primitives::hrp;
+#[doc(inline)]
 pub use crate::primitives::hrp::Hrp;
+#[doc(inline)]
 pub use crate::primitives::iter::{ByteIterExt, Fe32IterExt};
-pub use crate::primitives::{Bech32, Bech32m};
+#[doc(inline)]
+pub use crate::primitives::{Bech32, Bech32m, NoChecksum};
 
 mod error;
+/// Re-exports the hrp types from [`primitives::hrp`] to make importing ergonomic for the top level APIs.
+pub mod hrp;
```

Comment:
Love the re-exports to the top-level `hrp` module.

Do the doc comments need to point out that these are re-exports?

Reply:
There is a stupid compiler warning now if re-exports don't have doccomments. Since they appear in the docs under `Re-exports` with a link to their definition, this isn't really sensible, so we've been putting "placeholder comments" like this in rust-bitcoin.

Reply:
Commented on this below.